code_generation/license.h

```diff
@@ -83,6 +87,27 @@ class KODE_EXPORT License
      */
     QString text() const;
 
+    /**
+     * Returns license Type enum element by name. If there no such license,
+     * NoLicense will be returned
+     * @param license_string Name of license
+     * @return one of Type elements
+     */
+    static Type getLicenseNameByString(const QString &license_string);
```

Comment:
I would have made this a factory function instead. Why return an enum, rather than a License object?

    static License licenseByTypeName(const QString &typeName);

`Type type() const;` is missing for this to provide the same thing, of course. Should be added too.